SeaFightLogic/Ships/Ship.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Configuration;
+
+namespace SeaFight
+{
+
+    public abstract class Ship
+    {
+        public int Speed { get; protected set; }
+        public int ActionDistance { get; protected set; }
+        public int Size { get; protected set; }
+        public bool[] ShipCells { get; set; }
+
+        public Ship(int size, int speed)
+        {
+            int maxSize = Convert.ToInt32(ConfigurationManager.AppSettings["maxShipSize"]);
+            if (size < 1 || size > maxSize)
+            {
+                throw new ArgumentOutOfRangeException("Size of ship must be from 1 to 5");
+            }       
+            this.Speed = speed;
+
+            ShipCells = new bool[size];
+
+            for (int i = 0; i < ShipCells.Length; i++)
+            {
+                ShipCells[i] = true;
+            }
+
+            Size = size;
+        }
+
+        public void ShowState()
+        {
+            Console.WriteLine("Ship type: {0}", this.GetType().Name);

```

Comment:
It doesn't matter where current service can show the state

Reply:
Now return Dictionary<string,string>. But must replace with json object later